.buildkite/scripts/steps/artifacts/publish.sh

```diff
@@ -63,29 +63,29 @@ if [[ "$BUILDKITE_BRANCH" == "$KIBANA_BASE_BRANCH" ]]; then
   export BEATS_MANIFEST_URL=$(jq -r .manifest_url /tmp/beats_manifest.json)
 
   PUBLISH_CMD=$(cat << EOF
-  docker run --rm \
-    --name release-manager \
-    -e VAULT_ADDR \
-    -e VAULT_ROLE_ID \
-    -e VAULT_SECRET_ID \
-    --mount type=bind,readonly=false,src="$PWD/target",target=/artifacts/target \
-    docker.elastic.co/infra/release-manager:latest \
-      cli collect \
-        --project kibana \
-        --branch "$KIBANA_BASE_BRANCH" \
-        --commit "$GIT_COMMIT" \
-        --workflow "$WORKFLOW" \
-        --version "$BASE_VERSION" \
-        --qualifier "$VERSION_QUALIFIER" \
-        --dependency "beats:$BEATS_MANIFEST_URL" \
-        --artifact-set main
+docker run --rm \
+  --name release-manager \
+  -e VAULT_ADDR \
+  -e VAULT_ROLE_ID \
+  -e VAULT_SECRET_ID \
+  --mount type=bind,readonly=false,src="$PWD/target",target=/artifacts/target \
+  docker.elastic.co/infra/release-manager:latest \
+    cli collect \
+      --project kibana \
+      --branch "$KIBANA_BASE_BRANCH" \
+      --commit "$GIT_COMMIT" \
+      --workflow "$WORKFLOW" \
+      --version "$BASE_VERSION" \
+      --qualifier "$VERSION_QUALIFIER" \
+      --dependency "beats:$BEATS_MANIFEST_URL" \
+      --artifact-set main
 EOF
 )
   if [[ "${DRY_RUN:-}" =~ ^(1|true)$ ]]; then
-    PUBLISH_CMD+=(" --dry-run")
+    PUBLISH_CMD="$PUBLISH_CMD --dry-run"
   fi
 
-  "${PUBLISH_CMD[@]}"
+  eval "$PUBLISH_CMD"
```

Comment:
Do you think we can construct the flags and pass that to the docker command without eval at the end, or similar?  I'm hesitant to run eval against so many environment variables.

Reply:
oh, that's probably an option, I didn't consider that

Reply:
Without the eval, the command line seems to struggle with quotes, and I can't seem to nail the right way of setting the args up, see my attempts: https://buildkite.com/elastic/kibana-artifacts-snapshot/builds?branch=fix-publish-script

I've finally settled to have a very basic if/else branch here: 979b44a95e263515ea02926293b4c8c9985f69a5